Clients/GUI/Pages/UserInfo.razor

```diff
@@ -0,0 +1,133 @@
+            <label for="OldPassword">Enter old password: </label>
+            <InputText type="@(IsPasswordBeingChanged ? "password" : "hidden")" class="input-group-text" disabled="@(!IsPasswordBeingChanged)" @bind-Value="@user.OldPassword" />
+        </div>
+        <div class="form-group" style="@(IsPasswordBeingChanged ? "" : "height: 0; color: transparent")">
+            <label for="NewPassword">Enter new password: </label>
+            <InputText type="@(IsPasswordBeingChanged ? "password" : "hidden")" class="input-group-text" disabled="@(!IsPasswordBeingChanged)" @bind-Value="@user.NewPassword" />
+        </div>
+        <br />
+        <input type="@(IsEditModeDisabled ? "hidden" : "button")" @onclick="@ChangePasswordClickHandler" class="form-control col-3 button btn-primary" value="@(IsPasswordBeingChanged ? "Cancel changing password" : "Change password")" />
+        <br />
+        <ValidationSummary />
+        <h4 style="display:@(ErrorOccured ? "block" : "hidden"); color:red">@ErrorMessage</h4>
+        <input type="button" @onclick="@EditInfoClickHandler" class="form-control col-3 button btn-primary" value="@(IsEditModeDisabled ? "Edit" : "Save changes")" />
+    </EditForm>
+</div>
+
+@code {
+    public bool IsEditModeDisabled { get; set; }
+    public bool IsPasswordBeingChanged { get; set; }
+    public bool PopupVisible { get; set; }
+    public string ErrorMessage { get; set; }
+    public bool ErrorOccured { get; set; }
+    private UserInfoViewModel user;
+    private Guid currentId;
+
+    private void EditInfoClickHandler()
+    {
+        if (IsEditModeDisabled)
+        {
+            IsEditModeDisabled = false;
+        }
+        else // Trying to save changes
+        {
+            // Edit is inprocess in other team
+            try
+            {
+                var userInfo = new DTO.UserInformation(user.Name, user.Surname, user.DateOfBirth);
+
+                // This is in process (user/edit endpoint is another team's task)
+                if (IsPasswordBeingChanged)
+                {
+                    //UserEditor.EditUser(currentId, userInfo, user.OldPassword, user.NewPassword);
+                }
+                else
+                {
+                    //UserEditor.EditUser(currentId, userInfo, user.OldPassword, user.NewPassword);
+                }
+            }
+            // This must be changed to a custom exception later
+            catch (Exception)
+            {
+                // This will be changed to real error message
+                ErrorMessage = "error";
+                ErrorOccured = true;
+            }
+
+            Modal.Show<Modals.ChangesSavedModal>("Changes saved");
+            IsEditModeDisabled = true;
+            IsPasswordBeingChanged = false;
+        }
+    }
+
+    private void ChangePasswordClickHandler()
+    {
+        IsPasswordBeingChanged ^= true;
```

Comment:
What is this mean "^"

Reply:
https://en.wikipedia.org/wiki/Exclusive_or

Reply:
IsPasswordBeingChanged = IsPasswordBeingChanged ^ true. So, it is False if password was changed earlier and True if wasn't. 
Okay. So, how often we check this function? Each time when user put try to write something in password line?

Reply:
> IsPasswordBeingChanged = IsPasswordBeingChanged ^ true. So, it is False if password was changed earlier and True if wasn't.
> Okay. So, how often we check this function? Each time when user put try to write something in password line?

It's like a switch between showing fields for changing password and hiding them, so user has an option to change their password, they click the button and this triggers, showing the password change fields, and if user wants to cancel password changing process, they click button again, and property switches again.

Reply:
Thanks